django_project/frontend/src/containers/MainPage/Metrics/SpeciesLineChart.tsx

```diff
@@ -0,0 +1,175 @@
+import React, { useEffect, useState } from "react";
+import { Box, FormControl, MenuItem, Select, SelectChangeEvent, Typography } from "@mui/material";
+import { Line } from "react-chartjs-2";
+import { CategoryScale } from "chart.js";
+import Chart from "chart.js/auto";
+import "./index.scss";
+import PropertyInterface from "../../../models/Property";
+import axios from "axios";
+import Loading from "../../../components/Loading";
+import { useAppSelector } from "../../../app/hooks";
+import { RootState } from "../../../app/store";
+
+Chart.register(CategoryScale);
+
+const FETCH_PROPERTY_LIST_URL = '/api/property/list/'
+const FETCH_PROPERTY_POPULATION_SPECIES = '/species-population-count/'
+
+interface PopulationCount {
+    month: number;
+    month_total: number;
+}
+
+interface Species {
+    specie_name: string;
+    specie_colour: string;
+    annualpopulation_count: PopulationCount[];
+}
+
+const SpeciesLineChart = () => {
+    const selectedSpecies = useAppSelector((state: RootState) => state.SpeciesFilter.selectedSpecies)
+    const selectedMonths = useAppSelector((state: RootState) => state.SpeciesFilter.selectedMonths)
+    const startYear = useAppSelector((state: RootState) => state.SpeciesFilter.startYear)
+    const endYear = useAppSelector((state: RootState) => state.SpeciesFilter.endYear)
+    const [loading, setLoading] = useState(false)
+    const [selectedPropertyId, setSelectedPropertyId] = useState<number>(1)
+    const [propertyList, setPropertyList] = useState<PropertyInterface[]>([])
+    const [selectedProperty, setSelectedProperty] = useState<PropertyInterface>(propertyList?.[0])
```

Comment:
Why do you need both selectedPropertyId and selectedProperty? Could you derive selectedProperty from selectedPropertyId and propertyList?